CDAP-5645 http plugin cleanup

albertshau opened the pull request:
Renaming HttpCallback to HTTPCallback, and moving it into the
http-plugins module. Also renaming UrlFetch to HTTPPoller, so that
they have similar names, and consolidating common config logic for
the two plugins.

albertshau committed:
CDAP-5645 http plugin cleanup

Renaming HttpCallback to HTTPCallback, and moving it into the
http-plugins module. Also renaming UrlFetch to HTTPPoller, so that
they have similar names, and consolidating common config logic for
the two plugins.

awholegunch commented on http-plugins/docs/HTTPPoller-realtimesource.md:
```diff
@@ -19,28 +19,28 @@ Properties
         "type": "realtimesource",
-        "name": "UrlFetch",
         "properties": {
             "url": "http://api.example.com/sampleEndpoint",
```
Why not have the domain be "example.com"? Then the URL actually returns something, as that is what the `example.com` domain is intended for:

```
http://example.com/sampleEndpoint
```

Commit:
Update README.rst

Commit:
Update README.rst

awholegunch commented on http-plugins/src/main/java/co/cask/hydrator/plugin/batch/HttpCallbackAction.java:
```diff
@@ -114,14 +111,18 @@ public void run(BatchActionContext batchActionContext) throws Exception {
     @Nullable
+    @Description("When to run the action. Must be 'completion', 'success', or 'failure'. Defaults to 'completion'. " +
+      "If set to 'completion', the action will be executed regardless of whether " +
+      "the pipeline run succeeded or failed." +
```
Needs to end with a space:

```
"the pipeline run succeeded or failed. " +
```

russorat commented on http-plugins/README.rst:
```diff
@@ -5,31 +5,30 @@ URL Plugins
 ============
 
-A Hydrator Real Time Plugin that allows data to be fetched from a given URL at a specified interval.
+A Hydrator real-time plugin that allows data to be fetched from a given URL at a specified interval.
```
So i guess this needs to be rewritten a little bit since now this contains multiple plugins (and will contain more in the future). Maybe something like: 
This project is a collection of plugins to perform various HTTP actions in your pipeline. These plugins are currently available:
- HTTPPoller real-time Source
- HTTPCallback

albertshau replied:
I may actually just remove this readme.  There is one at the top level with almost all of this information (only non-overlap is the intro)

russorat commented on http-plugins/src/main/java/co/cask/hydrator/plugin/batch/HttpCallbackAction.java:
```diff
@@ -72,31 +72,28 @@ public void run(BatchActionContext batchActionContext) throws Exception {
         if (conf.body != null) {
           conn.setDoOutput(true);
           try (OutputStream outputStream = conn.getOutputStream()) {
             outputStream.write(conf.body.getBytes(Charsets.UTF_8));
```
Should the charset be configurable?

albertshau replied:
Since the body is a user supplied string, its being sent through rest as json and must be utf8.

russorat replied:
ok. is that a limit of hydrator or the plugin? maybe mention that it must be a json formatted string in the docs?

albertshau replied:
its just a property of json, since all the plugin configs are part of the app config in the create app request (http://docs.cdap.io/cdap/current/en/reference-manual/http-restful-api/lifecycle.html#create-an-application)

russorat replied:
ah, i keep forgetting this all gets translated into a json app config. sorry.

russorat commented on http-plugins/src/main/java/co/cask/hydrator/plugin/config/HTTPConfig.java:
```diff
@@ -0,0 +1,115 @@
+import javax.annotation.Nullable;
+
+/**
+ * Config class for URL Fetch plugin.
```
update comment

russorat commented on http-plugins/src/main/java/co/cask/hydrator/plugin/config/HTTPConfig.java:
```diff
@@ -0,0 +1,115 @@
+
+  @Description("Sets the connection timeout in milliseconds. Set to 0 for infinite. Default is 60000 (1 minute).")
+  @Nullable
+  private Integer connectTimeout;
```
Missing @Name annotation if that matters.

russorat commented on http-plugins/src/main/java/co/cask/hydrator/plugin/config/HTTPConfig.java:
```diff
@@ -0,0 +1,115 @@
+  @Nullable
+  private String requestHeaders;
+
+  @Name("followRedirects")
```
if at Name isn't required, this isn't needed either.

albertshau replied:
will remove, its not needed if its the same as the variable name.

russorat commented:
awesome rewrite! just a few minor comments.

albertshau committed:
CDAP-5645 small fixes

removing readme since the top level already has a readme, and
making small fixes to docs and removing unneeded annotation.

albertshau commented:
@russorat thanks for the review! Addressed the comments, please take another look

russorat commented:
👍  LGTM
